src/app/dataset-block/metadata-block/components/event-details/components/set-transform-event/set-transform-event.component.ts

```diff
@@ -0,0 +1,64 @@
+import { SetTransform } from "./../../../../../../api/kamu.graphql.interface";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ComponentRef,
+    Input,
+    QueryList,
+    ViewChildren,
+    ViewContainerRef,
+} from "@angular/core";
+import {
+    EventRow,
+    EventSection,
+    FACTORIES_BY_EVENT_TYPE,
+} from "../../builder.events";
+import { BasePropertyComponent } from "../common/base-property/base-property.component";
+
+@Component({
+    selector: "app-set-transform-event",
+    templateUrl: "./set-transform-event.component.html",
+    styleUrls: ["./set-transform-event.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SetTransformEventComponent {
```

Comment:
A lot of the logic can be reused from `SetPollingSourceComponent`. Introduce a base class and move everything shared there. Don't blindly copy&paste, think how to unify implementations as much as possible.

Reply:
Created one base component BaseEventComponent.